src/engine/Server.cpp

```diff
@@ -249,8 +249,9 @@ Awaitable<void> Server::process(
   // the key and value determines the kind of action, as well as parameters
   // like "index-decription=...", where the key determines the kind of action.
   auto checkParameter =
-      [&parameters](const std::string key, std::optional<std::string> value,
-                    bool accessAllowed = true) -> std::optional<std::string> {
+      [&parameters](
+          std::string_view key, std::optional<std::string_view> value,
+          bool accessAllowed = true) -> std::optional<std::string_view> {
```

Comment:
This can also be factored out into a free function that takes the `parameters` by reference as first argument, and is then here simply bound as
`auto checkParameter = std::bind_front(&Server::checkParameter, std::ref(parameters))`
(`std::bind_front` always copies by default. For this to work, you must specify the explicit type of the `parameters` argument and the `checkParameter` function must be static. Else you have to additionally `bind_front` the `*this`.